init/bash

```diff
@@ -1,41 +1,9 @@
-function show_msg {
-  # only echo msg if EESSI_SILENT is unset
-  msg=$1
-  if [[ -z ${EESSI_SILENT+x} ]]; then
-    echo "$msg"
-  fi
-}
-
-# The following method should be safe, but might break if file is a symlink
-# (could switch to $(dirname "$(readlink -f "$BASH_SOURCE")") in that case)
-source $(dirname "$BASH_SOURCE")/eessi_environment_variables
-
-# only continue if setting EESSI environment variables worked fine
-if [ $? -eq 0 ]; then
-
-    export PS1="{EESSI $EESSI_VERSION} $PS1"
-
-    # add location of commands provided by compat layer to $PATH;
-    # see https://github.com/EESSI/software-layer/issues/52
-    export PATH=$EPREFIX/usr/bin:$EPREFIX/bin:$PATH
-
-    # init Lmod
-    show_msg "Initializing Lmod..."
-    source $EESSI_EPREFIX/usr/share/Lmod/init/bash
-
-    # prepend location of modules for EESSI software stack to $MODULEPATH
-    show_msg "Prepending $EESSI_MODULEPATH to \$MODULEPATH..."
-    module use $EESSI_MODULEPATH
-    show_msg "Prepending site path $EESSI_SITE_MODULEPATH to \$MODULEPATH..."
-    module use $EESSI_SITE_MODULEPATH
-
-    #show_msg ""
-    #show_msg "*** Known problems in the ${EESSI_VERSION} software stack ***"
-    #show_msg ""
-    #show_msg "1) ..."
-    #show_msg ""
-    #show_msg ""
-
-    echo "Environment set up to use EESSI (${EESSI_VERSION}), have fun!"
-
-fi
+# Choose an EESSI version
+EESSI_VERSION="${EESSI_VERSION:-2023.06}"
+# Automatically load the module for EESSI version
+export LMOD_SYSTEM_DEFAULT_MODULES="${LMOD_SYSTEM_DEFAULT_MODULES:-EESSI/${EESSI_VERSION}}"
+# Path to top-level module tree
+export MODULEPATH=/cvmfs/software.eessi.io/versions/"$EESSI_VERSION"/init/modules
+. /cvmfs/software.eessi.io/versions/"$EESSI_VERSION"/compat/linux/$(uname -m)/usr/share/Lmod/init/bash
+
+module load "$LMOD_SYSTEM_DEFAULT_MODULES"
```

Comment:
The module load should not be necessary, they should be loaded when Lmod is initialised

Reply:
I'm not 100% though...https://lmod.readthedocs.io/en/latest/070_standard_modules.html#providing-a-standard-set-of-modules-for-all-users

Reply:
Was there already an Lmod loaded when you `source`d it?

`module load $LMOD_SYSTEM_DEFAULT_MODULES` is a little brittle as this won't work if there is more than module in the variable (I think). It's probably best to take the official route as per the Lmod docs (shell dependent):
```bash
if [ -z "$__Init_Default_Modules" ]; then
   export __Init_Default_Modules=1;

   ## ability to predefine elsewhere the default list
   LMOD_SYSTEM_DEFAULT_MODULES=${LMOD_SYSTEM_DEFAULT_MODULES:-"StdEnv"}
   export LMOD_SYSTEM_DEFAULT_MODULES
   module --initial_load --no_redirect restore
else
   module refresh
fi
```

Reply:
> Was there already an Lmod loaded when you `source`d it?

No, Lmod was not loaded at that moment. Only loaded with the script.